### PR TITLE
Share SqlConnectionTimeoutErrorInternal

### PR DESCRIPTION
**Pull request by Wraith2**
Another simple single file moved to shared. Minor style cleanup of the netcore version.

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj [hunk 1]
```diff
@@ -436,19 +436,31 @@
     <Compile Include="Microsoft\Data\SqlClient\SqlColumnEncryptionCngProvider.cs" />
     <Compile Include="Microsoft\Data\SqlClient\SqlColumnEncryptionCspProvider.cs" />
     <Compile Include="Microsoft\Data\SqlClient\SqlColumnEncryptionEnclaveProvider.cs" />
-    <Compile Include="Microsoft\Data\SqlClient\SqlCommand.cs" />
-    <Compile Include="Microsoft\Data\SqlClient\SqlCommandBuilder.cs" />
+    <Compile Include="Microsoft\Data\SqlClient\SqlCommand.cs">
+      <SubType>Component</SubType>
```
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj:440]: Cleanup "Component" changes

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlConnectionTimeoutErrorInternal.cs [hunk 2]
```diff
@@ -58,23 +60,21 @@ internal long GetMilliSecondDuration()
...
-        private SqlConnectionTimeoutPhaseDuration[] _originalPhaseDurations = null;
-
-        private SqlConnectionTimeoutErrorPhase _currentPhase = SqlConnectionTimeoutErrorPhase.Undefined;
-        private SqlConnectionInternalSourceType _currentSourceType = SqlConnectionInternalSourceType.Principle;
-        private bool _isFailoverScenario = false;
+        private SqlConnectionTimeoutPhaseDuration[] _phaseDurations;
+        private SqlConnectionTimeoutPhaseDuration[] _originalPhaseDurations;
+        private SqlConnectionTimeoutErrorPhase _currentPhase;
+        private SqlConnectionInternalSourceType _currentSourceType;
+        private bool _isFailoverScenario;
```
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlConnectionTimeoutErrorInternal.cs:67]: Any specific reason to remove these defaults?

Comment (Wraith2) [reply src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlConnectionTimeoutErrorInternal.cs:67]: They were the default/zero values for each type and the language requires that the be zero value on first access. Assigning to them at declaration point forces the creation of a ctor by the compiler. It was just unneeded noise.

File: src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj (hunk 1 above)
Comment (Wraith2) [reply src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj]: Done.

**Review by cheenamalhotra — APPROVED** [review]

**Review by johnnypham — APPROVED** [review]

**Comment by cheenamalhotra** [comment]
@Wraith2 there are some build issues in NetFx, please resolve them soon!
Thanks.

**Comment by Wraith2** [comment]
Fixed, thanks for pointing it out.

**Review by JRahnama — APPROVED** [review]

**Pull request merged**